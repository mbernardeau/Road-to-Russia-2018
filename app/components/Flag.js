import React, {
  Component,
} from 'react';
import PropTypes from 'prop-types';
import {
  firebaseConnect,
} from 'react-redux-firebase';

@firebaseConnect()
export default class Flag extends Component {
  static propTypes = {
    firebase: PropTypes.shape({
      storage: PropTypes.func.isRequired,
    }),
    country: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object,
  }

  constructor(props) {
    super(props);

    this.state = {
      url: null,
    };
  }

  componentDidMount() {
    this.storageRef = this.props.firebase.storage().ref();
    this.getUrl(this.props);
  }

  componentWillReceiveProps(nextProps) {
    this.getUrl(nextProps);
  }

  getUrl = (props) => {
    const { country } = props;

    if (country) {
      this.storageRef.child(`flags/${country}.svg`).getDownloadURL().then((url) => {
        this.setState({
          url,
        });
      });
    }
  }

  render() {
    const { url } = this.state;
    const { country, className, style } = this.props;

    return <img src={url} alt={country} className={className} style={style} />;
  }
}
