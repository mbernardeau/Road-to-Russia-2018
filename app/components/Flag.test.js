import React from 'react';
import renderer from 'react-test-renderer';

import Flag from './Flag';

describe('components/Flag', () => {
  jest.mock('assets/flags/ru.svg', () => 'mock_image');

  it('renders correctly', () => {
    const tree = renderer.create(
      <Flag country="ru" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should pass className and style to underlying component', () => {
    const tree = renderer.create(
      <Flag country="ru" style={{ marginTop: 4 }} className="test-image" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
