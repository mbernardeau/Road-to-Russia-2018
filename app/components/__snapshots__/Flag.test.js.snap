// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Flag renders correctly 1`] = `
<img
  alt="ru"
  className={undefined}
  src="mock_image"
  style={undefined}
/>
`;

exports[`components/Flag should pass className and style to underlying component 1`] = `
<img
  alt="ru"
  className="test-image"
  src="mock_image"
  style={
    Object {
      "marginTop": 4,
    }
  }
/>
`;
